app/components/post/post.js

```diff
@@ -486,175 +243,82 @@ class Post extends PureComponent {
+                        theme={theme}
+                        user={user}
                     />
                 </View>
-            );
-
-            displayName = (
-                <FormattedText
-                    id='post_info.system'
-                    defaultMessage='System'
-                    style={style.displayName}
-                />
-            );
-
-            messageStyle = [style.message, style.systemMessage];
-        } else if (post.props && post.props.from_webhook) {
-            if (config.EnablePostIconOverride === 'true') {
-                const icon = post.props.override_icon_url ? {uri: post.props.override_icon_url} : webhookIcon;
-                profilePicture = (
-                    <View style={style.profilePicture}>
-                        <Image
-                            source={icon}
-                            style={{
-                                height: PROFILE_PICTURE_SIZE,
-                                width: PROFILE_PICTURE_SIZE,
-                                borderRadius: PROFILE_PICTURE_SIZE / 2
-                            }}
+                <View style={style.messageContainerWithReplyBar}>
+                    {!commentedOnPost && this.renderReplyBar(style)}
+                    <View style={[style.rightColumn, (commentedOnPost && isLastReply && style.rightColumnPadding)]}>
+                        <PostHeader
+                            commentCount={commentCount}
+                            commentedOnDisplayName={commentedOnDisplayName}
+                            commentedOnPost={commentedOnPost}
+                            createAt={post.create_at}
+                            displayName={displayName}
+                            enablePostUsernameOverride={config.EnablePostUsernameOverride === 'true'}
+                            fromWebHook={post.props && post.props.from_webhook === 'true'}
+                            isPendingOrFailedPost={pendingOrFailed}
+                            isSystemMessage={isPostSystemMessage}
+                            onPress={this.handlePress}
+                            onViewUserProfile={this.viewUserProfile}
+                            overrideUsername={post.props && post.props.override_username}
+                            renderReplies={renderReplies}
+                            theme={theme}
+                        />
+                        <PostMessage
+                            attachments={post.props && post.props.attachments}
+                            canDelete={this.state.canDelete}
+                            canEdit={this.state.canEdit}
+                            fileIds={post.file_ids}
+                            hasBeenDeleted={post.state === Posts.POST_DELETED}
+                            intl={this.props.intl}
+                            isFlagged={this.props.isFlagged}
+                            isPendingOrFailedPost={pendingOrFailed}
```

Comment:
Going to have an issue here without distinguishing between pending and failed. The light background color is added for the post if it's pending or failed, but we only add the retry icon if it's failed. To see what I mean try uploading a file and you'll get a flash of the retry icon.